src/pages/index.js

```diff
@@ -59,57 +59,119 @@ const cardSection = new Section(
   selectors.cardSection
 );
 
+// Popup instances
 const imagePopup = new PopupWithImage(selectors.previewImagePopup);
+const profileEditPopup = new PopupWithForm(
+  selectors.profileEditPopup,
+  (formData) => {
+    return api
+      .updateUserInfo({
+        name: formData.title,
+        about: formData.description,
+      })
+      .then((updatedData) => {
+        userInfo.setUserInfo({
+          name: updatedData.name,
+          job: updatedData.about,
+        });
+        profileEditPopup.close();
+      })
+      .catch((err) => console.error("Error updating profile:", err));
+  }
+);
 
+const addCardPopup = new PopupWithForm(selectors.addCardPopup, (formData) => {
+  return api
+    .addCard({
+      name: formData.title,
+      link: formData.url,
+    })
+    .then((newCard) => {
+      const card = createCard(newCard);
+      cardSection.addItem(card);
+      addCardPopup.close();
+    })
+    .catch((err) => console.error("Error adding card:", err));
+});
+
+const deleteConfirmationPopup = new PopupWithConfirmation(
+  selectors.deleteCardPopup
+);
+deleteConfirmationPopup.setEventListeners();
+
+const avatarEditPopup = new PopupWithForm("#avatar-edit-popup", (formData) => {
+  return api
+    .updateAvatar(formData.avatar)
+    .then((updatedUserInfo) => {
+      userInfo.setUserInfo({
+        name: updatedUserInfo.name,
+        job: updatedUserInfo.about,
+        avatar: updatedUserInfo.avatar,
+      });
+      avatarEditPopup.close();
+    })
+    .catch((err) => console.error("Error updating avatar:", err));
+});
+avatarEditPopup.setEventListeners();
+
+// User Info instance
 const userInfo = new UserInfo({
   nameSelector: selectors.profileTitle,
   jobSelector: selectors.profileDescription,
+  avatarSelector: selectors.profileAvatar,
 });
 
-//initialize all instances
-
-cardSection.renderItems();
-
+// Initialize popups
 imagePopup.setEventListeners();
 profileEditPopup.setEventListeners();
 addCardPopup.setEventListeners();
+deleteConfirmationPopup.setEventListeners();
 
-//Event Listeners
+// Fetch initial data (user info and cards)
+api
+  .getInitialData()
+  .then(([userData, cards]) => {
```

Comment:
Good job that you get the whole info with `Promise.all`

Reply:
Thank you.